### PR TITLE
fix: cache fixes

### PR DESCRIPTION
**Pull request by ronag**
## Status

- [ ] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by ronag** [comment]
@flakey5

**Review by Uzlopak — COMMENTED** [review]
I love the direction this PR takes

File: types/cache-interceptor.d.ts [hunk 1]
```diff
@@ -31,10 +31,7 @@ declare namespace CacheHandler {
     path: string
   }
 
-  export interface GetResult {
-    response: CachedResponse
-    body?: Readable
-  }
+  type GetResult = CachedResponse & { body: null | Readable | Iterable<Buffer> | Buffer | Iterable<string> | string }
```
Comment (flakey5) [inline types/cache-interceptor.d.ts:34]: Could you add some more testing in the interceptor test file for each of the body types?

**Review by flakey5 — CHANGES_REQUESTED** [review]

File: types/cache-interceptor.d.ts (hunk 1 above)
Comment (ronag) [reply types/cache-interceptor.d.ts:34]: I'll change it back

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**